net-p2p/bitcoind/bitcoind-0.16.0.ebuild

```diff
@@ -0,0 +1,165 @@
+# Copyright 2010-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+DB_VER="4.8"
+inherit autotools bash-completion-r1 db-use systemd user
+
+MyPV="${PV/_/}"
+MyPN="bitcoin"
+MyP="${MyPN}-${MyPV}"
+BITCOINCORE_COMMITHASH="4b4d7eb255ca8f9a94b92479e6061d129c91a991"
+KNOTS_PV="${PV}.knots20180322"
+KNOTS_P="${MyPN}-${KNOTS_PV}"
+
+IUSE="+asm +bitcoin_policy_rbf examples knots libressl test upnp +wallet zeromq"
+
+DESCRIPTION="Original Bitcoin crypto-currency wallet for automated services"
+HOMEPAGE="https://bitcoincore.org/ https://bitcoinknots.org/"
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~amd64-linux ~arm ~arm64 ~mips ~ppc ~x86 ~x86-linux"
+
+SRC_URI="
+	https://github.com/${MyPN}/${MyPN}/archive/${BITCOINCORE_COMMITHASH}.tar.gz -> ${MyPN}-v${PV}.tar.gz
+	https://bitcoinknots.org/files/0.16.x/${KNOTS_PV}/${KNOTS_P}.patches.txz -> ${KNOTS_P}.patches.tar.xz
+"
+CORE_DESC="https://bitcoincore.org/en/2017/11/11/release-${PV}/"
+KNOTS_DESC="https://bitcoinknots.org/files/0.16.x/${KNOTS_PV}/${KNOTS_P}.desc.html"
+
+RDEPEND="
+	!libressl? ( dev-libs/openssl:0=[-bindist] )
+	libressl? ( dev-libs/libressl:0= )
+	dev-libs/libevent
+	>=dev-libs/libsecp256k1-0.0.0_pre20151118[recovery]
+	dev-libs/univalue
+	>=dev-libs/boost-1.52.0:=[threads(+)]
+	upnp? ( >=net-libs/miniupnpc-1.9.20150916 )
+	wallet? ( sys-libs/db:$(db_ver_to_slot "${DB_VER}")[cxx] )
+	zeromq? ( net-libs/zeromq )
+	virtual/bitcoin-leveldb
+"
+DEPEND="${RDEPEND}"
+
+DOCS=( doc/bips.md doc/files.md doc/reduce-traffic.md doc/release-notes.md )
+
+S="${WORKDIR}/${MyPN}-${BITCOINCORE_COMMITHASH}"
+
+pkg_pretend() {
+	if use knots; then
+		einfo "You are building ${PN} from Bitcoin Knots."
+		einfo "For more information, see ${KNOTS_DESC}"
+	else
+		einfo "You are building ${PN} from Bitcoin Core."
+		einfo "For more information, see ${CORE_DESC}"
+	fi
+	if use bitcoin_policy_rbf; then
+		einfo "Replace By Fee policy is enabled: Your node will preferentially mine and relay transactions paying the highest fee, regardless of receive order."
+	else
+		einfo "Replace By Fee policy is disabled: Your node will only accept the first transaction seen consuming a conflicting input, regardless of fee offered by later ones."
+	fi
+}
+
+pkg_setup() {
+	enewgroup bitcoin
+	enewuser bitcoin -1 -1 /var/lib/bitcoin bitcoin
+}
+
+KNOTS_PATCH() { echo "${WORKDIR}/${KNOTS_P}.patches/${KNOTS_P}.$@.patch"; }
```

Comment:
Remove this function and inline the results.

Reply:
Can I make a KNOTS_PATCHDIR variable? Simply inlining the entire thing makes it less readable IMO.

Reply:
Yes, please do that.